format_example.json

```diff
@@ -0,0 +1,135 @@
+{
+  "id": "111111-111111-111111-111111-111111",
+  "name": "Example Project",
+  "summary": "This is a short, one-sentence summary of the Example Project.",
+  "description": "This is a long description describing the project. <b>HTML</b> can optionally be used.",
+  "media": [
+    {
+      "rel": "icon",
+      "uri": "https://example.com/example.png",
+      "sha256": "< sum goes here >"
+    },
+    {
+      "type": "video",
+      "embed": "https://youtube.com/embed...",
+      "caption": "This is an example embed."
+    },
+    {
+      "type": "video",
+      "src": "https://example.com/video.webm",
+      "caption": "This is an example direct video link.",
+      "sha256": "< sum goes here >"
+    },
+    {
+      "type": "image",
+      "src": "https://example.com/image.png",
+      "caption": "This is an example image."
+    }
+  ],
+  "authors": [
+    {
+      "team": true,
+      "id": "111111-111111-111111-111111-111111",
+      "name": "MCIP Contributors",
+      "members": [
+        {
+          "id": "111111-111111-111111-111111-111111",
+          "name": "Example MCIP Member",
+          "uri": "https://example.com/example-webpage"
+        }
+      ]
+    }
+  ],
+  "categories": [
+    {
+      "name": "Example Category",
+      "uri": "https://example.com/browse?category=example",
+      "icon": "https://example.com/example-category.png"
+    }
+  ],
+  "license": {
+    "spdx": "MIT",
+    "uri": "https://opensource.org/licenses/MIT",
+    "modpackUsage": "alwaysAllow"
+  },
+  "links": [
+    {
+      "rel": [
+        "x-discord",
+        "community"
+      ],
+      "name": "Project Discord",
+      "uri": "https://discord.gg/example"
+    },
+    {
+      "rel": "scm",
+      "name": "Project Source",
+      "uri": "https://github.com/mc-cip/exampleproject"
+    }
+  ],
+  "successor": [
+    "successor-mod",
+    "other-successor-mod"
+  ],
+  "predecessor": "predecessor-mod",
+  "versions": [
+    {
+      "id": "1.0.0",
+      "name": "Release 1.0.0",
+      "semver": "1.0.0",
+      "releaseDate": "2020-04-08T12:00:00",
+      "changelog": "Example changelog goes here...",
+      "minecraftVersions": [
+        "1.16.1",
+        "1.16"
+      ],
+      "dependencies": [
+        {
+          "type": "asset",
+          "src": "example-host",
+          "id": "example-library-mod",
+          "required": true,
+          "version": [
+            "1.0.0",
+            "1.0.1"
+          ]
+        }
+      ],
+      "files": [
+        {
+          "rel": "primary",
+          "name": "1.0.0-fabric.jar",
+          "sha256": "< sum goes here >",
+          "dependencies": [
+            {
+              "type": "framework",
+              "id": "fabric-loader",
+              "version": ">= 0.6.1.45"
+            }
+          ],
+          "installationMethod": "placeInDirectory",
+          "directory": "mods",
```

Comment:
Should go inside installationMethod

Reply:
Yup, lower down also lists `"installationMethod": "modsDirectory",` so are they interchangeable?

Consistency would be good.

Reply:
I'll change `installationMethod` to an object and remove `modsDirectory`.

What other information could be added? Mentioned below was frameworks like Forge being provided as a project, so how could this be represented in `installationMethod`? Framework installation can be complicated, especially 1.13+ Forge.

Reply:
I suppose for Forge the installation method is forgeInstaller and for Fabric it's fabricInstaller. But this is not true for all versions! Old versions of Forge were just jar mods.

Reply:
forgeInstaller and fabricInstaller will probably be the best options. Launchers may choose to ignore the value as many don't run installers, but it's good to have the metadata there anyway.

Reply:
Fabric installation is fairly simple, all it really needs is a few extra libraries downloaded and a different mainClass, the installer just puts the vanilla launcher version json in the right place. The Fabric meta API has an endpoint for this https://meta.fabricmc.net/v2/versions/loader/1.16.1/0.9.0+build.204/profile/json. Forge installation since 1.13 pretty much requires an installer unless you use something like https://github.com/ZekerZhayard/ForgeWrapper (which just runs the installer on startup)

Reply:
Good point - should we include classpath dependencies? Can any mod have a classpath dependency, or just frameworks? Minecraft launcher wants a very specific format for classpath dependencies but theoretically any mod could want one, especially jar mods.

Reply:
Classpath dependencies are interesting - currently there's the `versionJsonInstall` installation method which tells launchers to look at the file with `rel` set to `versionJson`, which works for Fabric but it uses the Minecraft launcher's format.